src/python/WMCore/WMRuntime/Scripts/SetupCMSSWPset.py

```diff
@@ -5,190 +5,119 @@
+            self.__module = module
+            self.__name = name
+            self.__d = dict()
 
-    """
-    if not hasattr(process.source, "fileNames"):
-        process.source.fileNames = cms.untracked.vstring()
-    return
+        def __setitem__(self, key, value):
+            self.__d[key] = value
 
+        def __getitem__(self, item):
+            return self.__d[item]
 
-def fixupSecondaryFileNames(process):
-    """
-    _fixupSecondaryFileNames_
+        def __call__(self, *args, **kwargs):
+            pass
 
-    Make sure that the process has a secondaryFileNames parameter.
+        def __repr__(self):
+            return "{module}.{name}".format(module=self.__module, name=self.__name)
 
-    """
-    if not hasattr(process.source, "secondaryFileNames"):
-        process.source.secondaryFileNames = cms.untracked.vstring()
-    return
+    return DummyClass
 
 
-def fixupFirstLumi(process):
+class Unpickler(pickle.Unpickler):
     """
-    _fixupFirstLumi
+    _Unpickler_
+    Use this when loading a PSet pickle
+    and a specific module is not available (i.e.: FWCore modules)
 
-    Make sure that the process has firstLuminosityBlock parameter.
     """
-    if not hasattr(process.source, "firstLuminosityBlock"):
-        process.source.firstLuminosityBlock = cms.untracked.uint32(1)
-    return
+    def find_class(self, module, name):
+        return factory(module, name)
 
 
 class SetupCMSSWPset(ScriptInterface):
     """
     _SetupCMSSWPset_
 
     """
-    fixupDict = {"process.GlobalTag.globaltag": fixupGlobalTag,
-                 "process.GlobalTag.DBParameters.transactionId": fixupGlobalTagTransaction,
-                 "process.source.fileNames": fixupFileNames,
-                 "process.source.secondaryFileNames": fixupSecondaryFileNames,
-                 "process.maxEvents.input": fixupMaxEvents,
-                 "process.source.skipEvents": fixupSkipEvents,
-                 "process.source.firstEvent": fixupFirstEvent,
-                 "process.source.firstRun": fixupFirstRun,
-                 "process.source.lastRun": fixupLastRun,
-                 "process.source.lumisToProcess": fixupLumisToProcess,
-                 "process.source.firstLuminosityBlock": fixupFirstLumi}
 
     def __init__(self, crabPSet=False):
         ScriptInterface.__init__(self)
         self.crabPSet = crabPSet
         self.process = None
         self.jobBag = None
         self.logger = logging.getLogger()
+        self.tweak = PSetTweak()
+        self.scram = None
+        self.configPickle = "Pset.pkl"
+        self.psetFile = None
+
+    def createScramEnv(self):
+        """
+        _createScramEnv_
+
+        Create scram environment
+
+        """
+        scram = Scram(
+            version=self.getCmsswVersion(),
+            directory=self.stepSpace.location,
+            architecture=self.getScramVersion(),
+            initialise=self.step.data.application.setup.softwareEnvironment
+        )
```

Comment:
Can we add an extra log info level here reporting the CMSSW version and ScramArch being used?

Reply:
Yes, good idea.  I can add that.